x-pack/plugins/apm/public/components/app/ErrorGroupOverview/List/index.tsx

```diff
@@ -0,0 +1,149 @@
+`;
+
+const MessageAndCulpritCell = styled.div`
+  ${truncate('100%')};
+`;
+
+const MessageLink = styled(KibanaLink)`
+  font-family: ${fontFamilyCode};
+  font-size: ${fontSizes.large};
+  ${truncate('100%')};
+`;
+
+const Culprit = styled.div`
+  font-family: ${fontFamilyCode};
+`;
+
+export function ErrorGroupList({ items, urlParams, ...rest }: Props) {
+  const { serviceName, sortDirection, sortField } = urlParams;
+  const columns: ITableColumn[] = [
+    {
+      name: i18n.translate('xpack.apm.errorsTable.groupIdColumnLabel', {
+        defaultMessage: 'Group ID'
+      }),
+      field: 'groupId',
+      sortable: false,
+      width: px(unit * 6),
+      render: groupId => {
+        return (
+          <GroupIdLink hash={`/${serviceName}/errors/${groupId}`}>
+            {groupId.slice(0, 5) || NOT_AVAILABLE_LABEL}
+          </GroupIdLink>
+        );
+      }
+    },
+    {
+      name: i18n.translate(
+        'xpack.apm.errorsTable.errorMessageAndCulpritColumnLabel',
+        {
+          defaultMessage: 'Error message and culprit'
+        }
+      ),
+      field: 'message',
+      sortable: false,
+      width: '50%',
+      render: (message, item) => {
+        return (
+          <MessageAndCulpritCell>
+            <EuiToolTip
+              id="error-message-tooltip"
+              content={message || NOT_AVAILABLE_LABEL}
+            >
+              <MessageLink hash={`/${serviceName}/errors/${item.groupId}`}>
+                {message || NOT_AVAILABLE_LABEL}
+              </MessageLink>
+            </EuiToolTip>
+            <br />
+            <EuiToolTip
+              id="error-culprit-tooltip"
+              content={item.culprit || NOT_AVAILABLE_LABEL}
+            >
+              <Culprit>{item.culprit || NOT_AVAILABLE_LABEL}</Culprit>
+            </EuiToolTip>
+          </MessageAndCulpritCell>
+        );
+      }
+    },
+    {
+      name: '',
+      field: 'handled',
+      sortable: false,
+      align: 'right',
+      render: isUnhandled =>
+        isUnhandled === false && (
+          <EuiBadge color="warning">
+            {i18n.translate('xpack.apm.errorsTable.unhandledLabel', {
+              defaultMessage: 'Unhandled'
+            })}
+          </EuiBadge>
+        )
+    },
+    {
+      name: i18n.translate('xpack.apm.errorsTable.occurrencesColumnLabel', {
+        defaultMessage: 'Occurrences'
+      }),
+      field: 'occurrenceCount',
+      sortable: true,
+      dataType: 'number',
+      render: value =>
+        value ? numeral(value).format('0.[0]a') : NOT_AVAILABLE_LABEL
+    },
+    {
+      field: 'latestOccurrenceAt',
+      sortable: true,
+      name: i18n.translate(
+        'xpack.apm.errorsTable.latestOccurrenceColumnLabel',
+        {
+          defaultMessage: 'Latest occurrence'
+        }
+      ),
+      align: 'right',
+      render: value => (value ? moment(value).fromNow() : NOT_AVAILABLE_LABEL)
+    }
+  ];
+
+  return (
+    <ManagedTable
```

Comment:
Afaict this changes the behaviour, so errors are now sorted in-memory on the client - not in ES. 
If the user has 1000s of errors, and they try to sort by "occurrences" desc or "Latest occurrence" desc/asc they will now get incorrect results.

If we do decide to change this behaviour, we should also clean up all the server side sorting:
https://github.com/jasonrhodes/kibana/blob/e6590e7fcadc999f1786ddb4c33b62c3e4da62dd/x-pack/plugins/apm/server/lib/errors/get_error_groups.ts#L97-L107

It also doesn't look like `sortField` or `sortDirection` is used anymore:
https://github.com/jasonrhodes/kibana/blob/e6590e7fcadc999f1786ddb4c33b62c3e4da62dd/x-pack/plugins/apm/public/components/app/ErrorGroupOverview/List/index.tsx#L50

Reply:
Hmm, that's unfortunate that we can't consolidate this UX and then fix the client-side sorting issues once for everything, but we probably don't want to spread those issues before we have them fixed. I was hoping to avoid having to do lots of extra typing here. I'll revert.